src/systems/filecoin_blockchain/storage_power_consensus/storage_power_consensus_subsystem.go

```diff
@@ -86,23 +84,23 @@ func (spc *StoragePowerConsensusSubsystem_I) IsWinningPartialTicket(partialTicke
 	return spc.ec().IsWinningChallengeTicket(challengeTicket, activePowerInSector, totalPower)
 }
 
-func (spc *StoragePowerConsensusSubsystem_I) GetTicketProductionSeed(chain block.Chain, minerAddr addr.Address, epoch block.ChainEpoch) util.Randomness {
-	return chain.RandomnessAtEpoch(minerAddr, epoch-SPC_LOOKBACK_TICKET)
+func (spc *StoragePowerConsensusSubsystem_I) GetTicketProductionSeed(chain block.Chain, epoch block.ChainEpoch) util.Randomness {
+	return chain.RandomnessAtEpoch(epoch - SPC_LOOKBACK_TICKET)
 }
 
-func (spc *StoragePowerConsensusSubsystem_I) GetElectionProofSeed(chain block.Chain, minerAddr addr.Address, epoch block.ChainEpoch) util.Randomness {
-	return chain.RandomnessAtEpoch(minerAddr, epoch-SPC_LOOKBACK_RANDOMNESS)
+func (spc *StoragePowerConsensusSubsystem_I) GetElectionProofSeed(chain block.Chain, epoch block.ChainEpoch) util.Randomness {
+	return chain.RandomnessAtEpoch(epoch - SPC_LOOKBACK_RANDOMNESS)
 }
 
-func (spc *StoragePowerConsensusSubsystem_I) GetSealSeed(chain block.Chain, minerAddr addr.Address, epoch block.ChainEpoch) util.Randomness {
-	return chain.RandomnessAtEpoch(minerAddr, epoch-SPC_LOOKBACK_SEAL)
+func (spc *StoragePowerConsensusSubsystem_I) GetSealSeed(chain block.Chain, epoch block.ChainEpoch) util.Randomness {
+	return chain.RandomnessAtEpoch(epoch - SPC_LOOKBACK_SEAL)
 }
 
-func (spc *StoragePowerConsensusSubsystem_I) GetPoStChallenge(chain block.Chain, minerAddr addr.Address, epoch block.ChainEpoch) util.Randomness {
-	return chain.RandomnessAtEpoch(minerAddr, epoch-SPC_LOOKBACK_POST)
+func (spc *StoragePowerConsensusSubsystem_I) GetPoStChallenge(chain block.Chain, epoch block.ChainEpoch) util.Randomness {
+	return chain.RandomnessAtEpoch(epoch - SPC_LOOKBACK_POST)
 }
 
-func (spc *StoragePowerConsensusSubsystem_I) ValidateElectionProof(height block.ChainEpoch, electionProof block.ElectionProof, workerAddr addr.Address) bool {
+func (spc *StoragePowerConsensusSubsystem_I) ValidateElectionProof(height block.ChainEpoch, electionProof block.ElectionProof, minerAddr addr.Address) bool {
```

Comment:
The commented-out body of this function expects to obtain a public key for this address. Miner actor addresses have no such keys. If this is intended to be the address of the actor that signed something, it must be the worker address. Although passing the pubkey would be more useful.

Reply:
Yes! Sorry, this is all going away in a forthcoming PR to fix up EC for ePoSt so pay it no mind :)